### PR TITLE
[MI-3230]: Developed a modal which opens on click of view button in config table

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
## Summary

Developed a modal to open on click of view button in existing config table to receive and show the attachment  message

### CODE REVIEW

File: webapp/src/containers/components/modals/editActionsModal.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+import './styles.css';
+import Form from 'react-bootstrap/Form';
+
+interface Props {
+    visible: boolean;
+    setVis: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+function EditActionsModal(props: Props) {
```
Comment (Kshitij-Katiyar) [inline webapp/src/containers/components/modals/editActionsModal.tsx]: destructure the objects here only

File: webapp/src/containers/components/modals/editActionsModal.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    const [show, setShow] = useState(false);
+
+    useEffect(() => {
+        setShow(props.visible);
+    }, [props.visible]);
+
+    const handleClose = () => {
+        setShow(false);
+        props.setVis(false);
+    };
```
Comment (Kshitij-Katiyar) [inline webapp/src/containers/components/modals/editActionsModal.tsx]:
```suggestion
    };
    
```

File: webapp/src/containers/components/modals/editActionsModal.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                        <Form.Group>
+                            <Form.Label>{'Attachment Message'}</Form.Label>
+                            <Form.Control
+                                type='text'
+                                placeholder='Enter your attachment message'
+                            />
+                        </Form.Group>
+                        <Form.Group>
+                            <Form.Label>{'Action Type'}</Form.Label>
+                            <div className='dropdown'>
```
Comment (Kshitij-Katiyar) [inline webapp/src/containers/components/modals/editActionsModal.tsx]: fix the indentations everywhere

File: webapp/src/containers/components/modals/editActionsModal.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                                <Form.Select aria-label='Default select example'>
```
Comment (Kshitij-Katiyar) [inline webapp/src/containers/components/modals/editActionsModal.tsx]: why this aria-label='Default select example' ?

File: webapp/src/containers/components/modals/editActionsModal.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+                                    <option>{'Choose the type of your action'}</option>
+                                    <option value='1'>{'Button'}</option>
+                                    <option value='2'>{'Automatic'}</option>
+                                </Form.Select></div>
```
Comment (Kshitij-Katiyar) [inline webapp/src/containers/components/modals/editActionsModal.tsx]: fix identations

File: webapp/src/containers/components/modals/editActionsModal.tsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+                        onClick={handleClose}
+                    >{'Close'}</Button>
+                    <Button variant='primary'>{'Save changes'}</Button>
+                </Modal.Footer>
+            </Modal>
+        </>
+    );
+}
+
+export default EditActionsModal;
```
Comment (Kshitij-Katiyar) [inline webapp/src/containers/components/modals/editActionsModal.tsx]: Add EOF

File: webapp/src/containers/components/modals/styles.css [hunk 7]
```diff
@@ -0,0 +1,4 @@
+.modal {
+    opacity: 1;
+    padding-top: 10%;
+}
```
Comment (Kshitij-Katiyar) [inline webapp/src/containers/components/modals/styles.css]: same

File: webapp/src/containers/components/modals/viewActionsModal.tsx [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+                        variant='primary'
+                        disabled={true}
+                    >{'Save changes'}</Button>
+                </Modal.Footer>
+            </Modal>
+        </>
+    );
+}
+
+export default ViewActionsModal;
```
Comment (Kshitij-Katiyar) [inline webapp/src/containers/components/modals/viewActionsModal.tsx]: eof

File: webapp/src/containers/components/modals/editActionsModal.tsx [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+                                    <option>{'Choose the type of your action'}</option>
+                                    <option value='1'>{'Button'}</option>
+                                    <option value='2'>{'Automatic'}</option>
+                                </Form.Select></div>
+                        </Form.Group>
+                        <Form.Group>
+                            <Form.Label>{'Action Display Name'}</Form.Label>
+                            <Form.Control
+                                type='text'
+                                placeholder='Enter the display name of your action'
```
Comment (Kshitij-Katiyar) [inline webapp/src/containers/components/modals/editActionsModal.tsx]: remove your in every place holder like here Enter the display name of the action.

File: webapp/src/containers/components/modals/viewActionsModal.tsx [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+                    <Modal.Title>{'Actions'}</Modal.Title>
+                </Modal.Header>
+
+                <Modal.Body>
+                    <Form>
+                        <Form.Group>
+                            <Form.Label>{'Attachment Message'}</Form.Label>
+                            <Form.Control
+                                type='long-text'
+                                placeholder='This is a example attachment message'
```
Comment (Kshitij-Katiyar) [inline webapp/src/containers/components/modals/viewActionsModal.tsx]: change this

File: webapp/src/containers/components/tables/existingConfigTable.tsx [hunk 11]
```diff
@@ -25,15 +24,24 @@ type Props = {
     helpText: HelpText;
 }
 
-//TODO: Will remove this dummy data
 const ExistingConfigTable = ({label, helpText}: Props) => {
-    const dispatch = useDispatch();
+    const [vis, setVis] = useState(false);
```
Comment (Kshitij-Katiyar) [inline webapp/src/containers/components/tables/existingConfigTable.tsx]: bad naming conventions

File: webapp/src/containers/components/tables/existingConfigTable.tsx [hunk 12]
```diff
@@ -78,17 +90,22 @@ const ExistingConfigTable = ({label, helpText}: Props) => {
...
+                                            <Button
+                                                variant='primary'
+                                                onClick={handleView}
+                                            >
+                                                {'View'}
+                                            </Button>
+                                            <Button
+                                                variant='primary'
+                                                onClick={handleViews}
+                                            >{'Edit'}</Button>
```
Comment (Kshitij-Katiyar) [inline webapp/src/containers/components/tables/existingConfigTable.tsx]: identaions

File: webapp/src/containers/components/modals/editActionsModal.tsx [hunk 13]
```diff
@@ -0,0 +1,105 @@
+import React, {useEffect, useState} from 'react';
+import Button from 'react-bootstrap/Button';
+import Modal from 'react-bootstrap/Modal';
+
+import './styles.css';
+import Form from 'react-bootstrap/Form';
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/editActionsModal.tsx]: Fix import order

File: webapp/src/containers/components/modals/editActionsModal.tsx [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+
+interface Props {
+    visible: boolean;
+    setVis: React.Dispatch<React.SetStateAction<boolean>>;
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/editActionsModal.tsx]:
```suggestion
    setVisible: React.Dispatch<React.SetStateAction<boolean>>;
```

File: webapp/src/containers/components/modals/viewActionsModal.tsx [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+                                <td>{'Button'}</td>
+                                <td>{'Import'}</td>
+                                <td>{'channel1, channel2, channel3'}</td>
+                                <td>{'Welcome to your new team mate !'}</td>
+                            </tr>
+                            <tr>
+                                <td>{'Automatic'}</td>
+                                <td>{'Export'}</td>
+                                <td>{'channel1, channel2'}</td>
+                                <td>{'Welcome to your new team mate !'}</td>
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/viewActionsModal.tsx]:
```suggestion
                                <td>{'Welcome to your new team mate!'}</td>
```

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: webapp/src/containers/components/modals/editActionsModal.tsx [hunk 16]
```diff
@@ -0,0 +1,107 @@
+import React, {useEffect, useState} from 'react';
+
+import Button from 'react-bootstrap/Button';
+import Modal from 'react-bootstrap/Modal';
+import Form from 'react-bootstrap/Form';
+
+import './styles.css';
+
+
+interface Props {
```
Comment (avas27JTG) [inline webapp/src/containers/components/modals/editActionsModal.tsx]: use `types`

File: webapp/src/containers/components/modals/editActionsModal.tsx [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+    visible: boolean;
+    setVis: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+function EditActionsModal({visible, setVis}: Props) {
```
Comment (avas27JTG) [inline webapp/src/containers/components/modals/editActionsModal.tsx]: make it an arrow function

File: webapp/src/containers/components/modals/editActionsModal.tsx (hunk 6 above)
Comment (avas27JTG) [reply webapp/src/containers/components/modals/editActionsModal.tsx]: +1

File: webapp/src/containers/components/modals/styles.css (hunk 7 above)
Comment (avas27JTG) [reply webapp/src/containers/components/modals/styles.css]: +1

File: webapp/src/containers/components/modals/viewActionsModal.tsx [hunk 18]
```diff
@@ -0,0 +1,94 @@
...
+import Table from 'react-bootstrap/Table';
+import Form from 'react-bootstrap/Form';
+import './styles.css';
+
+interface Props {
+    visible: boolean;
+    setVisible: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+function ViewActionsModal(props: Props) {
```
Comment (avas27JTG) [inline webapp/src/containers/components/modals/viewActionsModal.tsx]: same as above fix for all such places

File: webapp/src/containers/components/modals/viewActionsModal.tsx [hunk 19]
```diff
@@ -0,0 +1,94 @@
...
+    useEffect(() => {
+        setShow(props.visible);
+    }, [props.visible]);
+
+    const handleClose = () => {
+        setShow(false);
+        props.setVisible(false);
+    };
+    return (
+        <>
```
Comment (avas27JTG) [inline webapp/src/containers/components/modals/viewActionsModal.tsx]: this is not needed as there is a single parent wrapper.

File: webapp/src/containers/components/modals/editActionsModal.tsx [hunk 20]
```diff
@@ -0,0 +1,107 @@
...
+    useEffect(() => {
+        setShow(visible);
+    }, [visible]);
+
+    const handleClose = () => {
+        setShow(false);
+        setVis(false);
+    };
+    return (
+        <>
```
Comment (avas27JTG) [inline webapp/src/containers/components/modals/editActionsModal.tsx]: this is not needed as there is a single parent wrapper.

File: webapp/src/containers/components/tables/existingConfigTable.tsx [hunk 21]
```diff
@@ -77,17 +92,21 @@ const ExistingConfigTable = ({label, helpText}: Props) => {
...
-                                            <Button variant='primary'>{'Edit'}</Button>
+                                            <Button
+                                                variant='primary'
+                                                onClick={handleView}
+                                            >
+                                                {'View'}
+                                            </Button>
+                                            <Button
+                                                variant='primary'
+                                                onClick={handleViews}
```
Comment (avas27JTG) [inline webapp/src/containers/components/tables/existingConfigTable.tsx]: change the name of `handleViews` to support this button `Edit`

File: webapp/src/containers/components/tables/existingConfigTable.tsx [hunk 22]
```diff
@@ -119,18 +143,26 @@ const ExistingConfigTable = ({label, helpText}: Props) => {
...
+                                            >
+                                                {'Edit'}
+                                            </Button>
```
Comment (avas27JTG) [inline webapp/src/containers/components/tables/existingConfigTable.tsx]: same

File: webapp/src/containers/components/modals/editActionsModal.tsx (hunk 14 above)
Comment (raghavaggarwal2308) [reply webapp/src/containers/components/modals/editActionsModal.tsx]: Why have you marked this as resolved if it is not fixed?

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: webapp/src/containers/components/modals/editActionsModal.tsx (hunk 14 above)
Comment (Kshitij-Katiyar) [reply webapp/src/containers/components/modals/editActionsModal.tsx]: fixed

File: webapp/src/containers/components/modals/styles.css [hunk 23]
```diff
@@ -1,4 +1,4 @@
-.show {
-    display: block;
-    position: initial;
+.modal {
+    opacity: 1;
+    padding-top: 10%;
```
Comment (avas27JTG) [inline webapp/src/containers/components/modals/styles.css]: why `%ge` and not `px`?

File: webapp/src/containers/components/modals/viewActionsModal.tsx [hunk 24]
```diff
@@ -0,0 +1,92 @@
...
+import Button from 'react-bootstrap/Button';
+import Modal from 'react-bootstrap/Modal';
+import Table from 'react-bootstrap/Table';
+import Form from 'react-bootstrap/Form';
+import './styles.css';
+
+interface Props {
+    visible: boolean;
+    setVisible: React.Dispatch<React.SetStateAction<boolean>>;
+}
```
Comment (avas27JTG) [inline webapp/src/containers/components/modals/viewActionsModal.tsx]: use `types`

File: webapp/src/containers/components/modals/viewActionsModal.tsx [hunk 25]
```diff
@@ -0,0 +1,92 @@
...
+
+    useEffect(() => {
+        setShow(props.visible);
+    }, [props.visible]);
+
+    const handleClose = () => {
+        setShow(false);
+        props.setVisible(false);
+    };
+    return (
```
Comment (avas27JTG) [inline webapp/src/containers/components/modals/viewActionsModal.tsx]:
```suggestion

    return (
```

File: webapp/src/containers/components/modals/styles.css (hunk 23 above)
Comment (Kshitij-Katiyar) [reply webapp/src/containers/components/modals/styles.css]: using % to keep the design responsive

**Review by raghavaggarwal2308 — APPROVED** [review]

File: webapp/src/containers/components/modals/styles.css (hunk 23 above)
Comment (avas27JTG) [reply webapp/src/containers/components/modals/styles.css]: It works fine here but % itself doesn't make the design responsive you can learn more about it to use it wisely.

**Review by avas27JTG — APPROVED** [review]

**Pull request merged**